pyam/core.py

```diff
@@ -458,6 +458,27 @@ def set_meta(self, meta, name=None, index=None):
         self._new_meta_column(name)
         self.meta[name] = meta[name].combine_first(self.meta[name])
 
+    def set_meta_from_data(self, name, method=None, column='value', **kwargs):
+        """Add meta indicators from downselected timeseries data
+
+        Parameters
+        ----------
+        name: str
+            meta column name
+        method: function, optional
+            method for aggregation, required if downselected data do not yield
+            unique values (e.g., `numpy.max()`)
+        column: str, optional
+            the column from `data` to be used for the meta indicator
+        kwargs: passed to :meth:`IamDataFrame.filter()` for downselected `data`
+        """
+        _data = self.filter(**kwargs).data
+        if method is None:
+            meta = _data.set_index(META_IDX)[column]
+        else:
+            meta = method(_data.groupby(META_IDX))[column]
```

Comment:
```suggestion
            meta = _data.groupby(META_IDX).apply(method)[column]
```

alternate way which might provide more flexibility

Reply:
Perhaps even `meta = _data.groupby(META_IDX)[column].apply(method)`?

Reply:
done